bricks/sync/waitable_atomic.h

```diff
@@ -214,17 +214,30 @@ class WaitableAtomic {
   // TODO(dkorolev): The `.Wait()` above always returning `true` could use some TLC.
 
   template <typename F>
-  std::invoke_result_t<F, data_t&> Wait(std::function<bool(const data_t&)> wait_predicate, F&& retval_predicate) {
+  std::invoke_result_t<F, data_t&> DoWait(std::function<bool(const data_t&)> wait_predicate, F&& retval_predicate) {
     std::unique_lock<std::mutex> lock(data_mutex_);
     if (!wait_predicate(data_)) {
       const data_t& data = data_;
-      data_condition_variable_.wait(lock, [&wait_predicate, &data] { return wait_predicate(data); });
+      data_condition_variable_.wait(lock, [&wait_predicate, &data]() { return wait_predicate(data); });
```

Comment:
CC @mzhurovich @amdrozdov -- I didn't know lambdas can be declared w/o `()` before, whoa!